### PR TITLE
Alerting/integrations error feedback handle 404 state not available

### PR DESCRIPTION
**Pull request by soniaAguilarPeiron**
**What this PR does / why we need it**:

This PR creates the new `fetchContactPointsState` using the actual endpoint `/api/alertmanager/${getDatasourceAPIUid(alertManagerSourceName)}/config/api/v1/receivers` handling 404 error when no state is available.

**Special notes for your reviewer**:
This PR adds tests for showing or not `NotificationError` in `Receivers` component.

### CODE REVIEW

File: public/app/features/alerting/unified/api/grafana.ts [hunk 1]
```diff
@@ -56,42 +61,73 @@ export const contactPointsStateDtoToModel = (receiversStateDto: ReceiversStateDT
 export const getIntegrationType = (integrationName: string): string | undefined =>
   parseIntegrationName(integrationName)?.type;
 
-export function fetchContactPointsState(alertManagerSourceName: String): Promise<ContactPointsState> {
-  return new Promise<ContactPointsState>((resolve) => {
+function isContactPointsStateNotAvailable(error: FetchError) {
+  return error.status === 404;
+}
+
+export async function fetchContactPointsState(alertManagerSourceName: string): Promise<ContactPointsState | undefined> {
```
Comment (gillesdemey) [inline public/app/features/alerting/unified/api/grafana.ts:68]: When would the promise return `undefined`?

File: public/app/features/alerting/unified/state/actions.ts [hunk 2]
```diff
@@ -456,7 +456,7 @@ export const fetchGrafanaNotifiersAction = createAsyncThunk(
 
 export const fetchContactPointsStateAction = createAsyncThunk(
   'unifiedalerting/fetchContactPointsState',
-  (alertManagerSourceName: String): Promise<ContactPointsState> =>
+  (alertManagerSourceName: string): Promise<ContactPointsState | undefined> =>
```
Comment (gillesdemey) [inline public/app/features/alerting/unified/state/actions.ts:459]: dito as above, when would this be `undefined`?

**Review by gillesdemey — APPROVED** [review]
LGTM couple of questions

File: public/app/features/alerting/unified/state/actions.ts (hunk 2 above)
Comment (soniaAguilarPeiron) [reply public/app/features/alerting/unified/state/actions.ts]: ops! true!! I've just pushed the change.

**Pull request merged**